### DISCUSSION TITLE
Working with categories and tags

### DISCUSSION THREAD

**Discussion by pmatulis** [mmistakes/minimal-mistakes#3059] [Q&A] [answered]
Hi. Right now I have category "files" (I'm not sure what to call them) in the root of my project. For example, in `docs/compute.md` I have:

```
---                                                                                                                                                            
title: Compute                                                                                                                                                 
layout: category                                                                                                                                               
permalink: /categories/compute/                                                                                                                                
taxonomy: compute                                                                                                                                              
entries_layout: list # or grid                                                                                                                                 
excerpt: "Something something"                                                                                                                                 
---
```

This works and I imagine it will for tags too, but what I'll end up with is a pile of files scattered throughout the root. Is it possible to have specific directories to put these files in (e.g. `_categories` and `_tags`)? I don't see any place where I can specify a path to these files.

Additionally, when I click on the resulting link at the bottom of a categorised (or tagged) post I get a directory listing from the internal webserver. Is that expected when testing locally?

![image](https://user-images.githubusercontent.com/6303808/125365192-cad0e800-e341-11eb-921b-daeac43e7b7a.png)

![image](https://user-images.githubusercontent.com/6303808/125365246-e6d48980-e341-11eb-9b83-2259bc8f856f.png)

In production I get a 404:

"Page not found:" https://example.com/categories/#compute

But this works fine:

https://example.com/categories/compute/

**Comment by mmistakes** [comment, marked as answer]
Part of the genius of `permalink` in front matter is whatever path you put in there, Jekyll will output the file there. So in your example `permalink: /categories/compute/` will output to `_site/categories/compute/index.html`

So what you can do is place these "categories" pages anywhere you want, keep them all in the root directory.

Put them all in a `/categories/` directory e.g.:

```
├── categories/
│   └── commute.md
│   └── some-other-category.md
```

Or if you like the Jekyll convention of having "source" directories begin with an underscore you could do:

```
├── _categories/
│   └── commute.md
│   └── some-other-category.md
```

But you'll need to add `_categories` to Jekyll's _config.yml under the `include` array. Any of the non-default underscore directories need to be included as Jekyll ignores them by defualt unless it's `_layouts`, `_includes`, `_sass` or `_posts`.

```yaml
# _config.yml

include:
  - .htaccess
  - _categories
  - _pages
```

The second part of your question, "Page not found:" https://example.com/categories/#compute
That file doesn't exist unless you're building a [`/categories/index.html`](https://mmistakes.github.io/minimal-mistakes/categories/) page that has all the categories on it. That's the default way the theme handles categories as it doesn't require manually building individual category pages.

But since you're creating those, you need to change the config slightly so it makes category links go to `/categories/category-name/` instead of `/categories/#category-name`. Check the docs and follow the settings for ["jekyll-archives"](https://mmistakes.github.io/minimal-mistakes/docs/configuration/#archive-settings). You're not using that plugin to create the category pages, but the URL structure will be the same.

**Reply by pmatulis** [reply]
1. Well I put `compute.md` under `/docs/categories` and it worked, even though I thought I tried this before. I still don't understand how it "finds" that file.

1. I went over the referred-to section and I'm afraid I'm going to need more help. My understanding is that I need to refer to the category using some variable in `config.yml`:

```
category_archive:
  type: liquid
  path: /categories/VARIABLE
```

**Reply by mmistakes** [reply]
Jekyll “finds” it because it’s a file with front matter. It doesn’t care where you put it.

If you didn’t set a `permalink` on that file it would output as `_site/docs/commute/`

And for the archive config just use what’s in the docs. It’s only changing the path of category/tag links. The default use #hashes, which you don’t want.

Try using these settings in _config.yml, you don’t need to add a variable at the end. That happens automatically.

```yaml
category_archive:
  type: jekyll-archives
  path: /categories/
tag_archive:
  type: jekyll-archives
  path: /tags/
```

**Reply by pmatulis** [reply]
1. OK
1. The ability to cite the uninstalled plugin was unexpected. Working. Thank you.

**Reply by pmatulis** [reply]
I now see that the 'Categories' component of the breadcrumbs fails in the same way. I don't suppose there is any way around that?

![image](https://user-images.githubusercontent.com/6303808/125488970-e0b66b7e-0a97-44c1-ac80-ebdd385e062c.png)

![image](https://user-images.githubusercontent.com/6303808/125489234-2eeedecf-0e76-46fb-bb27-090f94cce358.png)

**Reply by mmistakes** [reply]
They should respect the same Jekyll-archive settings. 

That said, The breadcrumb nav is technically in beta, and not something I suggest using. Unless your up for messing around with this include: https://github.com/mmistakes/minimal-mistakes/blob/master/_includes/breadcrumbs.html
